<!--REQUEST_ID=393c57a1-706a-4c53-8f18-a08f66b060b3-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48002&attempt_id=7ad0afd8-050d-41fc-a7e0-5b99b717f40e)

Attempted: cups-googlecloudprint

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/zyyw7qxpj1yri8xdasbm68ix3hwmly85-coreutils-8.30/bin/install -c -m 755 ./testing/listdrivefiles.py /nix/store/8nrc60szgg5ghxpmyc8nd8qlsjbpfgka-cups-googlecloudprint-20160502/share/cloudprint-cups/testing/
/nix/store/zyyw7qxpj1yri8xdasbm68ix3hwmly85-coreutils-8.30/bin/install -c -m 644 ./testing/testfiles/* /nix/store/8nrc60szgg5ghxpmyc8nd8qlsjbpfgka-cups-googlecloudprint-20160502/share/cloudprint-cups/testing/testfiles/
/nix/store/zyyw7qxpj1yri8xdasbm68ix3hwmly85-coreutils-8.30/bin/install -c -m 644 ./selinux/* /nix/store/8nrc60szgg5ghxpmyc8nd8qlsjbpfgka-cups-googlecloudprint-20160502/share/cloudprint-cups/selinux/
/nix/store/zyyw7qxpj1yri8xdasbm68ix3hwmly85-coreutils-8.30/bin/install -c -m 644 ./oauth2client/*.py /nix/store/8nrc60szgg5ghxpmyc8nd8qlsjbpfgka-cups-googlecloudprint-20160502/share/cloudprint-cups/oauth2client/
mkdir -p /Library/LaunchDaemons/
/nix/store/zyyw7qxpj1yri8xdasbm68ix3hwmly85-coreutils-8.30/bin/install -c -m 644 ./launchd/* /Library/LaunchDaemons/
install: cannot create regular file '/Library/LaunchDaemons/cupscloudprint.plist': Permission denied
make: *** [Makefile:34: install] Error 1
builder for '/nix/store/ly5gbqashma0x14fw6rrbsgz8dqgnl8h-cups-googlecloudprint-20160502.drv' failed with exit code 2
error: build of '/nix/store/ly5gbqashma0x14fw6rrbsgz8dqgnl8h-cups-googlecloudprint-20160502.drv' failed
```
</p></details>

--- a/nixos/modules/services/printing/cupsd.nix
+++ b/nixos/modules/services/printing/cupsd.nix
@@ -250,7 +250,7 @@ in
       drivers = mkOption {
         type = types.listOf types.path;
         default = [];
-        example = literalExample "[ pkgs.gutenprint pkgs.hplip pkgs.splix ]";
+        example = literalExample "with pkgs; [ gutenprint hplip splix cups-googlecloudprint ]";
         description = ''
           CUPS drivers to use. Drivers provided by CUPS, cups-filters,
           Ghostscript and Samba are added unconditionally. If this list contains

--- a/pkgs/misc/cups/drivers/googlecloudprint/default.nix
+++ b/pkgs/misc/cups/drivers/googlecloudprint/default.nix
@@ -1,0 +1,64 @@
+{ stdenv, lib, fetchFromGitHub, python2, python2Packages, file, makeWrapper, cups }:
+
+# Setup instructions can be found at https://github.com/simoncadman/CUPS-Cloud-Print#configuration
+# So the nix version is something like:
+# nix run nixpkgs.cups-googlecloudprint -c sudo setupcloudprint
+# nix run nixpkgs.cups-googlecloudprint -c sudo listcloudprinters
+
+let pythonEnv = python2.buildEnv.override {
+  extraLibs = with python2Packages; [
+    six
+    httplib2
+    pycups
+  ];
+};
+
+in stdenv.mkDerivation rec {
+  name    = "cups-googlecloudprint-${version}";
+  version = "20160502";
+
+  src = fetchFromGitHub {
+    owner  = "simoncadman";
+    repo   = "CUPS-Cloud-Print";
+    rev    = version;
+    sha256 = "0760i12w7jrhq7fsgyz3yqla5cvpjb45n6m2jz96wsy3p3xf6dzz";
+  };
+
+  buildInputs = [ cups makeWrapper ];
+
+  cupsgroup = "nonexistantgroup";
+  NOPERMS = 1;
+
+  postConfigure = ''
+    substituteInPlace Makefile --replace "${cups}" "$out"
+  '';
+
+  postInstall = ''
+    pushd "$out"
+    for s in lib/cups/backend/gcp lib/cups/driver/cupscloudprint
+    do
+      echo "Wrapping $s..."
+      wrapProgram "$out/$s" --set PATH "${lib.makeBinPath [pythonEnv file]}" --prefix PYTHONPATH : "$out/share/cloudprint-cups"
+    done
+
+    mkdir bin
+
+    for s in share/cloudprint-cups/*.py
+    do
+      if [ -x "$s" ] # Only wrapping those Python scripts marked as executable
+      then
+        o="bin/$(echo $s | sed 's,share/cloudprint-cups/\(.*\).py,\1,')"
+        echo "Wrapping $o -> $s..."
+        makeWrapper "$out/$s" "$o" --set PATH "${lib.makeBinPath [pythonEnv file]}" --prefix PYTHONPATH : "$out/share/cloudprint-cups"
+      fi
+    done
+    popd
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Google Cloud Print driver for CUPS, allows printing to printers hosted on Google Cloud Print";
+    homepage    = http://ccp.niftiestsoftware.com;
+    platforms   = platforms.linux;
+    license     = licenses.gpl3;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -21606,6 +21606,8 @@ with pkgs;
 
   cups-brother-hl1110 = pkgsi686Linux.callPackage ../misc/cups/drivers/hl1110 { };
 
+  cups-googlecloudprint = callPackage ../misc/cups/drivers/googlecloudprint { };
+
   # this driver ships with pre-compiled 32-bit binary libraries
   cnijfilter_2_80 = pkgsi686Linux.callPackage ../misc/cups/drivers/cnijfilter_2_80 { };
 

